src/OrchardCore.Modules/OrchardCore.Queries/Sql/Controllers/AdminController.cs

```diff
@@ -79,32 +83,27 @@ public async Task<IActionResult> Query(AdminQueryViewModel model)
             var dialect = _store.Configuration.SqlDialect;
 
             var parameters = JsonConvert.DeserializeObject<Dictionary<string, object>>(model.Parameters);
-
             var tokenizedQuery = await _liquidTemplateManager.RenderStringAsync(model.DecodedQuery, NullEncoder.Default, parameters.Select(x => new KeyValuePair<string, FluidValue>(x.Key, FluidValue.Create(x.Value, _templateOptions))));
+            var dialect = _store.Configuration.SqlDialect;
 
             if (SqlParser.TryParse(tokenizedQuery, _store.Configuration.Schema, dialect, _store.Configuration.TablePrefix, parameters, out var rawQuery, out var messages))
             {
                 model.RawSql = rawQuery;
                 model.Parameters = JsonConvert.SerializeObject(parameters, Formatting.Indented);
-
                 try
                 {
-                    using (connection)
-                    {
-                        await connection.OpenAsync();
-                        model.Documents = await connection.QueryAsync(rawQuery, parameters);
-                    }
+                    model.Documents = await _queryExecutor.QueryAsync<dynamic>(rawQuery, parameters);
```

Comment:
You kept for example line 82

            await using var connection = _store.Configuration.ConnectionFactory.CreateConnection();